ambry-store/src/main/java/com/github/ambry/store/StoreMetrics.java

```diff
@@ -250,6 +262,7 @@ private void deregisterIndexGauges(String storeId) {
     registry.remove(MetricRegistry.name(Log.class, prefix + "CurrentCapacityUsed"));
     registry.remove(MetricRegistry.name(Log.class, prefix + "PercentageUsedCapacity"));
     registry.remove(MetricRegistry.name(Log.class, prefix + "CurrentSegmentCount"));
+    registry.remove(MetricRegistry.name(Log.class, prefix + "CurrentInvalidDataSize"));
```

Comment:
just being curious, why are we removing these metrics here? It's just that I've never seen we remove gauge metrics  anywhere else.

Reply:
it's for test purpose. metric can't be registered twice. In test, store may start twice, so we remove it. `CurrentSegmentCount ` is an existing example.

Reply:
it's weird to do such thing, to register a gauge, we can call `MetricRegistry.gauge`, this way, we can register same metric multiple times ( this method would check if it's already registered or not). `MetricRegistry.register` should be avoided. This way, we don't have to remove metrics from registry.

Reply:
You made a good point. 

But for this change, we probably need to use `MetricRegistry.register` because BlobStore may be shutdown due to replica move, so we need to de-register some metrics.